compiler/rustc_builtin_macros/src/deriving/decodable.rs

```diff
@@ -149,7 +149,20 @@ fn decodable_substructure(
                 arms.push(cx.arm(v_span, cx.pat_lit(v_span, cx.expr_usize(v_span, i)), decoded));
             }
 
-            arms.push(cx.arm_unreachable(trait_span));
+            let expr_unreachable = cx.expr_macro_call(
```

Comment:
nit: it probably makes sense to keep `arm_unreachable` as a function and just put this in it to keep the code simpler?

Reply:
I re-added some of the methods that were initially removed.